### ISSUE TITLE
Detect indirect deadlock through thread return and join

### ISSUE THREAD

**Issue by sim642** [goblint/analyzer#667]
In https://github.com/goblint/analyzer/pull/655/commits/3124e376b93054e742b6cd302e5c71bf8ccf2949 @michael-schwarz added a deadlock example, where a thread forgets to unlock a mutex before returning. Then the joining thread deadlocks trying to lock it.

Detecting this is more involved because it's not just a locking ordering issue. And it might not fall under the classical deadlock definition of multiple threads waiting for resources, but might be nice to still warn about.

One obvious idea would be more along the lines of #176 to just check the must/may locksets at thread returns and warn about them being non-empty since that's already unusual. Although not necessarily indicative of an actual deadlock.

A more precise idea might be to add additional edges into the deadlock analysis graph corresponding to such thread returns and joins. Then it might be possible to arrange things such that the above situation also causes a cycle, but through other edges than simply locking order.

**Comment by sim642** [comment]
The man page for `pthread_join` also says:
```
ERRORS
       EDEADLK
              A deadlock was detected (e.g., two threads tried to join with each other); or thread specifies the calling thread.
```
So there can be deadlocks through just joins as well.
